### DISCUSSION TITLE
Arduino Nano RP2040 and ST7735 from AZ-Delivery

### DISCUSSION THREAD

**Discussion by kww52** [Bodmer/TFT_eSPI#1350] [Compatible displays and setup files]
Hello,
Is there a way to get the ST7735 from az-delivery to work with the Arduino Nano RP2040.
https://www.az-delivery.de/products/1-8-zoll-spi-tft-display
It works fine with the Adafruit GFX and Adafruit ST7735.
Unfortunately, with the TFT_eSPI library, I only get a blank screen.

#define TFT_MISO 12
#define TFT_MOSI 11
#define TFT_SCLK 13
#define TFT_CS 10
#define TFT_DC 8
#define TFT_RST 9

What am I doing wrong?
[Setup60_RP2040_ST7735.h.txt](https://github.com/Bodmer/TFT_eSPI/files/7252791/Setup60_RP2040_ST7735.h.txt)

Best Regards
kww52

**Comment by ruifig** [comment]
Probably the same issue being discussed in https://github.com/Bodmer/TFT_eSPI/discussions/1519
Try defining the pins as their GPIO equivalent with the `PinName` enum. For your case, probably this

```
#define TFT_MISO p4
#define TFT_MOSI p7
#define TFT_SCLK p6
#define TFT_CS p5
#define TFT_DC p20
#define TFT_RST p21
```

**Comment by aord** [comment]
Did you get this SPI working? What connections did you use?

In the Setup60:
"With a ST7735 display more than 27MHz may not work (spurious pixels and lines)."
You are using #define SPI_FREQUENCY  70000000 (70MHz)
perhaps try #define SPI_FREQUENCY   1000000 or other 

I have the same screen and like you on the Pico Pi, I also have just a white screen. I'm not sure my pin connection is correct and can't find a detailed pin connection list some has used. Not sure if I should be using SPI1 or SPI0?

Using the RP2040 MBed board manager SPI works with Adafruit ST7735 (however I can't get any MBed multicore to work :(  ).

**Reply by aord** [reply]
I just saw Bodmer wrote: "You must use #define TFT_SPI_PORT 1 since you are using pins 11, 12, 13". for another display so I guess it is the same for the ST7735.
